php/stats.php

```diff
@@ -39,101 +41,121 @@
+$sql = <<<SQL
+SELECT COUNT(DISTINCT case when alid <> -1 then alid end) AS num_airlines,
+       COUNT(DISTINCT plid) AS num_planes,
+       COUNT(*) as segments,
+       IFNULL(SUM(distance), 0) AS distance,
+       IFNULL(AVG(distance), 0) AS avg_distance,
+       IFNULL(SUM(TIME_TO_SEC(duration)), 0) AS duration,
+       IFNULL(TIME_FORMAT(SEC_TO_TIME(SUM(TIME_TO_SEC(duration))/COUNT(duration)), '%H:%i'), '00:00') AS avg_duration
 FROM flights AS f
-WHERE $filter";
+WHERE $filter
+SQL;
 $sth = $dbh->query($sql);
 $row = $sth->fetch();
 if ($row) {
-    $array += $row;
+    $response["unique"]["carriers"] = (int)$row["num_airlines"];
+    $response["unique"]["vehicles"] = (int)$row["num_planes"];
+    $response["total"]["segments"] = (int)$row["segments"];
+    $response["total"]["distance_mi"] = round((int)$row["distance"]);
+    $response["total"]["distance_km"] = round((int)$row["distance"] * KM_PER_MILE);
+    $response["total"]["distance"] = round((int)$row["distance"] * $unitMultiplier);
+    $response["total"]["duration_s"] = round((int)$row["distance"] * KM_PER_MILE);
+    $response["average"]["distance"] = round((int)$row["avg_distance"] * $unitMultiplier);
+    $response["average"]["duration"] = $row["avg_duration"];
 }
-$array["avg_distance"] = number_format($array["avg_distance"]) . " " . $unit;
-$array["localedist"] = number_format(round($array["distance"] * ($units == "K" ? KM_PER_MILE : 1))) . " " . $unit;
-print json_encode($array) . "\n";
 
 // longest and shortest
 // 0 desc, 1 distance, 2 duration, 3 src_iata, 4 src_icao, 5 src_apid, 6 dst_iata, 7 dst_icao, 8 dst_apid
 $sql = <<<SQL
 (
-    SELECT '%s', ROUND(f.distance %s) AS distance, DATE_FORMAT(duration, '%%H:%%i') AS duration,
-           s.iata, s.icao, s.apid, d.iata, d.icao, d.apid
+    SELECT 'longest' AS prop, f.distance, DATE_FORMAT(duration, '%H:%i') AS duration,
+           s.iata AS src_iata, s.icao AS src_icao, s.apid AS src_apid,
+           d.iata AS dst_iata, d.icao AS dst_icao, d.apid AS dst_apid
     FROM flights AS f, airports AS s, airports AS d
     WHERE f.src_apid = s.apid AND f.dst_apid = d.apid AND $filter
     ORDER BY distance DESC
     LIMIT 1
 )
-UNION 
+UNION
 (
-    SELECT '%s', ROUND(f.distance %s) AS distance, DATE_FORMAT(duration, '%%H:%%i') AS duration,
-           s.iata, s.icao, s.apid, d.iata, d.icao, d.apid
+    SELECT 'shortest' AS prop, f.distance, DATE_FORMAT(duration, '%H:%i') AS duration,
+           s.iata AS src_iata, s.icao AS src_icao, s.apid AS src_apid,
+           d.iata AS dst_iata, d.icao AS dst_icao, d.apid AS dst_apid
     FROM flights AS f, airports AS s, airports AS d
     WHERE f.src_apid = s.apid AND f.dst_apid = d.apid AND $filter
     ORDER BY distance ASC
     LIMIT 1
 )
 SQL;
-$sth = $dbh->query(
-    sprintf(
-        $sql,
-        _("Longest"),
-        $multiplier,
-        _("Shortest"),
-        $multiplier
-    )
-);
-$rows = [];
+$sth = $dbh->query($sql);
```

Comment:
https://www.php.net/manual/en/pdo.query.php

We should pass `PDO::FETCH_ASSOC` as the second parameter, as we don't need the 0 based indexed columns anymore.